core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/branch/Branch.java

```diff
@@ -68,36 +69,64 @@
 		 * @throws BadRequestException
 		 */
 		void checkName(String name) throws BadRequestException;
+		
+		/**
+		 * Returns the name from a temporary branch name if it is one, otherwise returns the branch name.
+		 * 
+		 * @param name
+		 * @throws BadRequestException
+		 */
+		String getName(String name) throws BadRequestException;
 
 		/**
 		 * @since 4.2
 		 */
 		class BranchNameValidatorImpl implements BranchNameValidator {
 
-			private Pattern pattern;
-			private String allowedCharacterSet;
-			private int maximumLength;
-
+			private final Pattern branchNamePattern;
+			
+			private final Pattern tmpBranchNamePattern;
+			
+			
 			public BranchNameValidatorImpl() {
-				this(DEFAULT_ALLOWED_BRANCH_NAME_CHARACTER_SET, DEFAULT_MAXIMUM_BRANCH_NAME_LENGTH);
+				branchNamePattern = Pattern.compile(String.format("^(%s)?[%s]{1,%s}(_[0-9]{1,19})?$",
+						Pattern.quote(TEMP_PREFIX),
+						DEFAULT_ALLOWED_BRANCH_NAME_CHARACTER_SET,
+						DEFAULT_MAXIMUM_BRANCH_NAME_LENGTH));
+				
+				tmpBranchNamePattern = Pattern.compile(String.format("^(%s)([%s]{1,%s})(_[0-9]{1,19}$)",
+						Pattern.quote(Branch.TEMP_PREFIX), 
+						Branch.DEFAULT_ALLOWED_BRANCH_NAME_CHARACTER_SET, 
+						Branch.DEFAULT_MAXIMUM_BRANCH_NAME_LENGTH));
 			}
-
-			public BranchNameValidatorImpl(String allowedCharacterSet, int maximumLength) {
-				this.allowedCharacterSet = allowedCharacterSet;
-				this.maximumLength = maximumLength;
-				pattern = Pattern.compile(String.format("^(%s)?[%s]{1,%s}(_[0-9]{1,19})?$", Pattern.quote(TEMP_PREFIX), allowedCharacterSet, maximumLength));
-			}
-
+					
 			@Override
 			public void checkName(String name) {
 				if (Strings.isNullOrEmpty(name)) {
 					throw new BadRequestException("Name cannot be empty");
 				}
-				if (!pattern.matcher(name).matches()) {
+				if (!branchNamePattern.matcher(name).matches()) {
 					throw new BadRequestException(
 							"'%s' is either too long (max %s characters) or it contains invalid characters (only '%s' characters are allowed).", name,
-							maximumLength, allowedCharacterSet);
+							DEFAULT_MAXIMUM_BRANCH_NAME_LENGTH, DEFAULT_ALLOWED_BRANCH_NAME_CHARACTER_SET);
+				}
+			}
+			
+			@Override
+			public String getName(String name) {
+				checkName(name);
+				if (name.startsWith(TEMP_PREFIX)) {
+					final Matcher matcher = tmpBranchNamePattern.matcher(name);
+					if (matcher.matches()) { 
+						return matcher.group(2);
+					} else  {
+						throw new BadRequestException(String.format("%s didn't match for pattern.", name)); 
```

Comment:
Please change the message to something like "' < name > ' is not a valid temporary branch name format"